src/mame/pc/ps2m50.cpp

```diff
@@ -0,0 +1,198 @@
+}
+
+uint8_t ps2_m50_t1_mb_device::planar_pos_r(offs_t offset)
+{
+	LOGPOS("%s: O:%02X\n", FUNCNAME, offset);
+
+	switch(offset)
+	{
+		case MCABus::POS::ADAPTER_ID_LO: return m_planar_id & 0xFF;
+		case MCABus::POS::ADAPTER_ID_HI: return (m_planar_id & 0xFF00) >> 8;
+		case MCABus::POS::OPTION_SELECT_DATA_1: return m_io_controller->system_board_io_r();
+
+		// TODO: Figure out what the memory bits are - type 1 can only ever have 1MB onboard (2x512K 30-pin)
+		//		 this seems to get it seeing 1MB
+		case MCABus::POS::OPTION_SELECT_DATA_2: return 0b01010100 | m_system_board_memory_enabled;
+		default: return 0x00;
+	}
+}
+
+void ps2_m50_t1_mb_device::planar_pos_w(offs_t offset, uint8_t data)
+{
+	switch(offset)
+	{
+		case 2:
+		// System Board I/O Byte.
+			LOGPOS("PS/2 system board I/O enable bits now %02X\n", data);
+			m_io_controller->system_board_io_w(data);
+			break;
+		case 3:
+		// Memory Control Register
+			LOGPOS("PS/2 memory control register now %02X\n", data);
+			update_memory_control_register(data);
+			break;
+		default:
+			LOGPOS("Unhandled planar POS write: reg %02X data %02X\n", offset, data);
+	}
+}
+
+void ps2_m50_t1_mb_device::update_memory_control_register(uint8_t data)
+{
+    m_system_board_memory_enabled = BIT(data, 0);
+
+    // The system board enable/disable bit controls the low 640K.
+    if(m_system_board_memory_enabled)
+    {
+        m_maincpu->space(AS_PROGRAM).install_ram(0x000000, 0x09ffff, m_ram->pointer()); 
+    }
+    else
+    {
+        m_maincpu->space(AS_PROGRAM).unmap_readwrite(0x000000, 0x09ffff);
+    }
+}
+
+/****************************************
+ * Model 60
+ * 
+ * - Basically a Model 50 in a tower case.
+ * - 8 slots means it gains 2KB of SRAM.
+ * - Same BIOS ROM as the Model 50.
+ ****************************************/
+
+void ps2_m60_mb_device::device_start()
+{
+	ps2_m50_t1_mb_device::device_start();
+
+	m_sram = make_unique_clear<uint8_t[]>(m_sram_size);
+	m_nvram->set_base(m_sram.get(), m_sram_size);
+	save_pointer(NAME(m_sram), m_sram_size);
+}
+
+void ps2_m60_mb_device::device_reset()
+{
+	ps2_m50_t1_mb_device::device_reset();
+}
+
+void ps2_m60_mb_device::device_add_mconfig(machine_config &config)
+{
+	ps2_mb_device::device_add_mconfig(config);
+
+	add_mca16(config);
+    add_dmac_72x7377(config);
+	add_southbridge_72x8299(config);
+
+	MCA16_SLOT(config, m_mcaslot[0], 0, m_mcabus, pc_mca16_cards, nullptr, false);
+	MCA16_SLOT(config, m_mcaslot[1], 0, m_mcabus, pc_mca16_cards, nullptr, false);
+	MCA16_SLOT(config, m_mcaslot[2], 0, m_mcabus, pc_mca16_cards, nullptr, false);
+	MCA16_SLOT(config, m_mcaslot[3], 0, m_mcabus, pc_mca16_cards, nullptr, false);
+	MCA16_SLOT(config, m_mcaslot[4], 0, m_mcabus, pc_mca16_cards, nullptr, false);
+	MCA16_SLOT(config, m_mcaslot[5], 0, m_mcabus, pc_mca16_cards, nullptr, false);
+	MCA16_SLOT(config, m_mcaslot[6], 0, m_mcabus, pc_mca16_cards, nullptr, false);
+	MCA16_SLOT(config, m_mcaslot[7], 0, m_mcabus, pc_mca16_cards, nullptr, false);
+}
```

Comment:
Please check your IDE settings, a bunch of these new files don't have a newline at end of file.